kraken-java-sdk/kraken-java-sdk-gateway/src/main/java/com/consoleconnect/kraken/operator/gateway/runner/MappingMatrixCheckerActionRunner.java

```
@@ -277,6 +315,74 @@ public boolean checkExpect(PathCheck pathCheck, Object value) {
     return false;
   }
 
+  public void checkModifyConstraints(
+      List<FilterRule> filterRules, String targetKey, Map<String, Object> inputs) {
+    if (CollectionUtils.isEmpty(filterRules)) {
+      return;
+    }
+    // Reading the configurable items to compare
+    Optional<UnifiedAsset> modifyAssetOpt = readFromPath(getAppProperty().getModifyUseCase());
+    if (modifyAssetOpt.isEmpty()) {
+      return;
+    }
+    UnifiedAsset modifyAsset = modifyAssetOpt.get();
+    ComponentValidationFacets modifyFacets =
+        UnifiedAsset.getFacets(modifyAsset, ComponentValidationFacets.class);
+    Set<String> supportedUseCases =
+        modifyFacets.getModificationRules().stream()
+            .map(ComponentValidationFacets.ModificationRule::getUseCase)
+            .collect(Collectors.toSet());
+    UnifiedAssetDto targetAsset = unifiedAssetService.findOne(targetKey);
+    String mapperKey = targetAsset.getMetadata().getMapperKey();
+    if (!supportedUseCases.contains(mapperKey)) {
+      return;
+    }
+    Optional<ComponentValidationFacets.ModificationRule> modificationRuleOpt =
+        modifyFacets.getModificationRules().stream()
+            .filter(item -> item.getUseCase().equals(mapperKey))
+            .findFirst();
+    if (modificationRuleOpt.isEmpty()) {
+      return;
+    }
+    ComponentValidationFacets.ModificationRule modificationRule = modificationRuleOpt.get();
+    // Read instance id
+    DocumentContext documentContext = JsonPath.parse(inputs);
+    // 400 if no parameter
+    FilterRule filterRule = filterRules.get(0);
+    Object instanceObj =
+        readByPathWithException(documentContext, filterRule.getQueryPath().trim(), List.of(), "");
+    // Query Order payload by instance id
+    if (Objects.isNull(instanceObj)) {
+      return;
```

should return when instanceId is null. Better to throw an exception here

---

Thanks a lot! An exception of 400 should be thrown here.